packages/react-native-reanimated/src/createAnimatedComponent/PropsFilter.tsx

```diff
@@ -58,21 +58,15 @@ export class PropsFilter implements IPropsFilter {
         // it will help other libs to interpret styles correctly
         props[key] = processedStyle;
       } else if (key === 'animatedProps') {
-        const animatedPropsProp = inputProps.animatedProps;
-        const animatedPropsArray = flattenArray<
-          Partial<AnimatedComponentProps<AnimatedProps>>
-        >(animatedPropsProp ?? []);
-
-        animatedPropsArray.forEach((animatedProps) => {
-          if (animatedProps?.viewDescriptors && animatedProps.initial) {
-            Object.keys(animatedProps.initial.value).forEach(
-              (initialValueKey) => {
-                props[initialValueKey] =
-                  animatedProps.initial?.value[initialValueKey];
-              }
-            );
-          }
-        });
+        const animatedProp = inputProps.animatedProps as Partial<
+          AnimatedComponentProps<AnimatedProps>
+        >;
+        if (animatedProp.initial !== undefined) {
+          Object.keys(animatedProp.initial.value).forEach((initialValueKey) => {
+            props[initialValueKey] =
+              animatedProp.initial?.value[initialValueKey];
+          });
+        }
```

Comment:
I think I'm missing something because I don't understand why in the base branch you're handling `animatedProp` as an array to allow passing multiple animated props objects, but here you're handling `animatedProp` as an object instead of an array 🤔.